Track archive

tim-hogan opened the pull request:
Tidy up of the tracks database and management of tracks.

dan-corneanu commented on includes/classGlidingDB.php:
```diff
@@ -0,0 +1,170 @@
+<?php
+require_once dirname(__FILE__) . '/classSQLPlus.php';
+class GlidingDB extends SQLPlus 
```
@tim-hogan where is the definition of `SQLPlus` ? does it come from a library or did you write it? I can not find it. Could it be that you forgot to add it to git?

tim-hogan replied:
I forgot to load, now loaded

tim-hogan commented on includes/classTracksDB.php:
```diff
@@ -0,0 +1,30 @@
+<?php
+require_once './includes/classSQLPlus.php';
```
Done

tim-hogan commented on includes/classGlidingDB.php:
```diff
@@ -0,0 +1,170 @@
+    //*********************************************************************
+    public function getAircraft($id)
+    {
+        return $this->singlequery("SELECT * from aircraft where id = " . intval($id));
```
All good, but I have need used and therfore did not want to stray into the unkown.

tim-hogan commented on includes/classTracksDB.php:
```diff
@@ -0,0 +1,30 @@
+    //*********************************************************************
+    // Tracks
+    //*********************************************************************  
+    public function numTracksForFlight($start,$end,$aircraft)
```
Let me check to see where else its used in the helper.  If I cannot find anywhere else I will reaplce remove from helper also, as thats is very old code.

tim-hogan replied:
It is still used in MyFlights and the igc generator.  
Best option will be to leave as buggy now, and I will patch the above 2 as a separye bug as I need to change the mapping with another branch anyway.

dan-corneanu replied:
Yes, it is still used but it can be replaced
e.g.
in igcgenerate.php:46
```
if (tracksforFlight($con,null,$glider,$trDateStart->format('Y-m-d H:i:s'),$trDateLand->format('Y-m-d H:i:s')) )
```
can be replaced by
```
if(GlidingDB->numTracksForFlight(....) > 0)
```
and on line 52
```
if (tracksforFlight(null,$con2,$glider,$trDateStart->format('Y-m-d H:i:s'),$trDateLand->format('Y-m-d H:i:s')) )
```
can be replaced by
```
if(TracksDB->numTracksForFlight(....) >0)
```


Am I wrong?

tim-hogan replied:
You are correct.  Havn't tested by yes this woudl work.  Was going o be my next rewrite, but as igs=c is very raely used, was concentraing on the problems we had expereinced with archive first.
Tim

dan-corneanu replied:
Maybe I am missing something, but everywhere `tracksforFlight` is called, it is used like
```
if (tracksforFlight(....)) {
  either load somthing from gliding or archives database
}
```
so I think we can replace all calls to `itracksforFlight` with either a call to `GlidingDB->numTracksForFlight` or `TracksDB->numTracksForFlight` depending on which DB is referenced inside the `if` block (gliding or archives)

tim-hogan replied:
Correct, but I havn't gone through them all with  double check yet.
I suggest I add to this branch the otherchnages and remove from helpers.
My point was to do that later, and simply hurry to get tracks working, but I will therfore delay and fix inother places.  Let you know when completed,

dan-corneanu replied:
> My point was to do that later, and simply hurry to get tracks working, but I will therfore delay and fix inother places. Let you know when completed,

I am sorry for the mess I have created with that buggy helper. 
If in a hurry then we can absolutely merge this branch and address the bug in another change.
Absolutely fine with me.

tim-hogan commented:
I think I have now finished so this pull can be checked and implemented,

dan-corneanu commented on MyFlightMap.php:
```diff
@@ -71,53 +77,36 @@
+$trDateLand = new DateTime($to);
+
+$r = null;
+if ($DB->numTracksForFlight($trDateStart,$trDateLand,$glider) > 0)
```
👍

dan-corneanu commented on MyFlights.php:
```diff
@@ -287,7 +296,7 @@ function printit(){window.print();}
   $trDateLand->setTimestamp(intval(floor($row[10])));
 
-  if (tracksforFlight($con,$con2,$row[1],$trDateStart->format('Y-m-d H:i:s'),$trDateLand->format('Y-m-d H:i:s')) )
+  if ($DB->numTracksForFlight($trDateStart,$trDateLand,$row[1]) > 0 || $DBArchive->numTracksForFlight($trDateStart,$trDateLand,$row[1]) > 0)
```
nice

dan-corneanu commented:
I have approved your changes. Thanks for getting rid of *tracksforFlight*.
You can now merge this branch into master and deploy into production.

I have noticed once again that there are modified files on glidingops.com
and also there are files that are not tracked.
This is really confusing and gets in your way when deploying.

modified:   MyFlights.php
modified:   apiglidjsonv1.php
modified:   igcgenerate.php

untracked files:
GlidingFlightMap.php
MyFlights-save.php
apiglidjsonv1-save.php
igcgenerate-save.php
includes/
testDB.php

If they need to be on the server they should be in github and deployed from
there.
My idea was that *glidingops.com <http://glidingops.com>* is always clean
and it has the latest master branch from github deployed on it. If I am
working on something, I make changes in a branch, I test them on my
computer and when happy and PR is reviewed and approved, I merge into
master and deploy on *glidingops.com <http://glidingops.com>* It should be
possible to test everything on local machine without moving files into
glidingops.com until ready for production.

Let me know what you think.

Regards,
Dan


On Fri, Mar 1, 2019 at 5:03 PM tim-hogan <notifications@github.com> wrote:

> I think I have now finished so this pull can be checked and implemented,
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/45#issuecomment-468534625>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADK1hlYeekpx1gcIeC996li2_-2kcY_Mks5vSKaFgaJpZM4bOP_o>
> .
>

tim-hogan commented:
The Modeifed are not actuall, they were saved as one name and then saved back again, whist I test the new versions.
So they are in fact the originals , but many have a different tiem stamp.

As for.

GlidingFlightMap.php

New and still in test for 3D Flight maps.

MyFlights-save.php
apiglidjsonv1-save.php
igcgenerate-save.php

Simply saved old versions

includes/testDB.php

Old test.

So this should all come right after the branch.

After I have merged the branch is there an auto install?

Timbo



From: Dan Corneanu <notifications@github.com>
Sent: Friday, 1 March 2019 10:15 PM
To: tim-hogan/glidingops <glidingops@noreply.github.com>
Cc: Tim Hogan <tim@mobilelocate.co.nz>; Mention <mention@noreply.github.com>
Subject: Re: [tim-hogan/glidingops] Track archive (#45)

I have approved your changes. Thanks for getting rid of *tracksforFlight*.
You can now merge this branch into master and deploy into production.

I have noticed once again that there are modified files on glidingops.com
and also there are files that are not tracked.
This is really confusing and gets in your way when deploying.

modified: MyFlights.php
modified: apiglidjsonv1.php
modified: igcgenerate.php

untracked files:
GlidingFlightMap.php
MyFlights-save.php
apiglidjsonv1-save.php
igcgenerate-save.php
includes/
testDB.php

If they need to be on the server they should be in github and deployed from
there.
My idea was that *glidingops.com <http://glidingops.com>* is always clean
and it has the latest master branch from github deployed on it. If I am
working on something, I make changes in a branch, I test them on my
computer and when happy and PR is reviewed and approved, I merge into
master and deploy on *glidingops.com <http://glidingops.com>* It should be
possible to test everything on local machine without moving files into
glidingops.com until ready for production.

Let me know what you think.

Regards,
Dan


On Fri, Mar 1, 2019 at 5:03 PM tim-hogan <notifications@github.com<mailto:notifications@github.com>> wrote:

> I think I have now finished so this pull can be checked and implemented,
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/tim-hogan/glidingops/pull/45#issuecomment-468534625>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADK1hlYeekpx1gcIeC996li2_-2kcY_Mks5vSKaFgaJpZM4bOP_o>
> .
>


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/tim-hogan/glidingops/pull/45#issuecomment-468597537>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ATRwg30XP50dx60JbA6g0v2iwhGIm4H6ks5vSO9-gaJpZM4bOP_o>.
